contents/dataframes_performance.md

```diff
@@ -4,8 +4,237 @@ So far, we didn't think about making our `DataFrames.jl` code **fast**.
 Like everything in Julia, `DataFrames.jl` can be really fast.
 In this section, we will give some performance tips and tricks.
 
-```{=comment}
-- Functions with a shebang versus without (e.g., filter! versus filter)
-- df[!, col] versus df[:, col]
-- other things? Jose probably knows some more
+### Allocations
+
+Like we explained in @sec:function_bang functions that end with a bang `!` are a common pattern to denote functions that modify one or more of their arguments.
+In the context of high performance Julia code, this *means* that **functions with `!` do not perform any sort of allocation**.
+They just change in place any object that we supply as arguments.
+
+Almost all the `DataFrames.jl` functions that we've seen so far have a \"`!` twin\".
+For example, `filter` has a non allocating `filter!`, `select` has `select!`, `subset` has `subset!`, and so on.
+Notice that these functions **do not** return a `DataFrame`, but instead they **alter** the `DataFrame` that they act upon.
+Additionally, there are some functions that do not have a `!` counterpart.
+For example, all the `join`s, by logical purposes, *cannot* be done without allocations.
+Thus, we don't have any `join!` functions in `DataFrames.jl`.
+
+If you want the highest speed and performance in your code you should definitely use the `!` functions instead of regular `DataFrames.jl` functions.
```

Comment:
```suggestion
If you want the highest speed and performance in your code, you should definitely use the `!` functions instead of regular `DataFrames.jl` functions.
```
"You should definitely use [...]" is a sentence which can stand on its own. Therefore, there should be a comma.